processing/src/main/java/org/apache/druid/math/expr/Function.java

```diff
@@ -3724,6 +3724,44 @@ ExprEval doApply(ExprEval arrayExpr, ExprEval scalarExpr)
     }
   }
 
+  class ArrayScalarInFunction extends ArrayScalarFunction
```

Comment:
I wonder if it would be possible to instead of adding a new native function - delegate all this work to the new type-aware native in filter @clintropolis have added?